resize-window.el

```diff
@@ -177,166 +271,369 @@ nil."
-to enlarge right."
+Press n to resize down, p to resize up, b to resize left and f
+to resize right."
   (interactive)
-  (setq resize-window--background-overlay (resize-window--make-background))
-  (resize-window--notify "Resize mode: enter character, ? for help")
+  (resize-window--refresh-stack)
+  ;; NOTE: Do not trim the stack here. Let stack requests to handle
+  ;; window configurations in excess.
+  (resize-window--add-backgrounds)
+  (resize-window--notify-status)
   (condition-case nil
-   (let ((reading-characters t)
-         ;; allow mini-buffer to collapse after displaying menu
-         (resize-mini-windows t))
-     (while reading-characters
-       (let* ((char (resize-window--match-alias (read-key)))
-              (choice (assoc char resize-window-dispatch-alist))
-              (capital (when (numberp char)
-                         (assoc (+ char 32) resize-window-dispatch-alist))))
-         (cond
-          (choice (resize-window--execute-action choice))
-          ((and capital (resize-window--allows-capitals capital))
-           ;; rather than pass an argument, we tell it to "scale" it
-           ;; with t and that method can worry about how to get that
-           ;; action
-           (resize-window--execute-action capital t))
-          (t (setq reading-characters nil)
-             (delete-overlay resize-window--background-overlay))))))
-   (quit (resize-window--delete-overlays))))
+      (let ((reading-keys t)
+            ;; allow mini-buffer to collapse after displaying menu
+            (resize-mini-windows t))
+        (while reading-keys
+          (let* ((kin (read-key-sequence-vector nil nil t))
+                 (key (and kin (resize-window--match-alias kin)))
+                 (choice (and key (resize-window--match-dispatch key)))
+                 (lower (and key (resize-window--key-to-lower key)))
+                 (capital (and lower (resize-window--match-dispatch lower))))
+            (cond
+             (choice (resize-window--execute-action choice))
+             ((and capital (resize-window--allows-capitals capital))
+              ;; rather than pass an argument, we tell it to "scale" it
+              ;; with t and that method can worry about how to get that
+              ;; action
+              (resize-window--execute-action capital t))
+             ((or (resize-window--keys-equal key [?q])
+                  (resize-window--keys-equal key [?Q])
+                  (resize-window--keys-equal key [? ])
+                  (resize-window--keys-equal key "C-g"))
+              (setq reading-keys nil)
+              (resize-window--cancel-notify)
+              (resize-window--display-menu 'kill)
+              (resize-window--remove-backgrounds))
+             (t
+              (resize-window--notify
+               (format
+                "Unregistered key: %s -> %s"
+                key (resize-window--key-str key))))))))
+    (quit
+     (resize-window--display-menu 'kill)
```

Comment:
this doesn't call `resize-window--cancel-notify` like when a `[?q]` input is read. Is this deliberate? Seems like cleanup should largely be consistent.

Reply:
You just spot a bug! I'll fix it.

Reply:
Fixed in forced push.